### PR TITLE
Remove re-exports

### LINKED ISSUE #852: Get rid of redundant re-exports
Concerned files:

```
schedule/mod.rs
messages/mod.rs
treesync/mod.rs
```

### PR DESCRIPTION
**Pull request by raphaelrobert**
Fixes #852.

This PR removes re-exports (in particular on the crate-level).

### CODE REVIEW

File: openmls/src/extensions/mod.rs [hunk 1]
```diff
@@ -22,25 +22,28 @@
...
 
+// Private
 mod capabilities_extension;
-mod errors;
 mod external_key_id_extension;
 mod life_time_extension;
 mod parent_hash_extension;
 mod ratchet_tree_extension;
 mod required_capabilities;
+use errors::*;
```
Comment (kkohbrok) [inline openmls/src/extensions/mod.rs:40]: Should this line not be above the `// Private` line?

File: openmls/src/key_packages/mod.rs [hunk 2]
```diff
@@ -95,10 +95,10 @@ use tls_codec::{
 
 // Private
 mod codec;
-mod errors;
+use errors::*;
```
Comment (kkohbrok) [inline openmls/src/key_packages/mod.rs:98]: Same here.

File: openmls/src/messages/mod.rs [hunk 3]
```diff
@@ -15,45 +15,36 @@ use crate::{
...
-    types::{CryptoError, HpkeCiphertext},
+    types::{Ciphersuite, CryptoError, HpkeCiphertext},
     OpenMlsCryptoProvider,
 };
 use serde::{Deserialize, Serialize};
 use thiserror::Error;
 
 // Private
-mod codec;
+use proposals::*;
```
Comment (kkohbrok) [inline openmls/src/messages/mod.rs:25]: and here.

**Review by kkohbrok — CHANGES_REQUESTED** [review]
Thanks for the going through all of that! Just some places where the `use` statements should be moved up a few lines, I think. I pointed out some of them, but I might have missed one or two.

File: openmls/src/extensions/mod.rs (hunk 1 above)
Comment (raphaelrobert) [reply openmls/src/extensions/mod.rs:40]: I put the `use` of submodules in the private section. the other `use` statements are for things that are no submodules.

**Review by kkohbrok — APPROVED** [review]

**Pull request merged**